internal/common/regions.go

```diff
@@ -1,23 +1,112 @@
 package common
 
 import (
+	"context"
+	"fmt"
+	"strings"
+
+	"log"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	"github.com/aws/aws-sdk-go-v2/service/sts"
 	"github.com/aws/aws-sdk-go/aws/endpoints"
 )
 
-func GetAWSRegions(selectedRegions []string) []string {
+func GetAWSRegions(ctx context.Context, cfg aws.Config, selectedRegions []string) ([]string, error) {
+	log.Println("Starting GetAWSRegions function")
+
+	var regionsToCheck []string
 	if len(selectedRegions) > 0 {
-		return selectedRegions
+		regionsToCheck = selectedRegions
+		log.Printf("Using selected regions: %v\n", regionsToCheck)
```

Comment:
Don't use `log`. We wrap everything with `svc1log` and the Palantir witchcraft go logging. You'll want to do something like:
```
log := svc1log.FromContext(cmd.Context())
log.Info(fmt.Sprintf(...))
```